project-koku/koku
# [COST-2445] Move OCPGCP ui table summary from trino to postgres

Fixes [COST-2445](https://issues.redhat.com/browse/COST-2445)

Changes proposed in this PR:
* OCP on GCP UI Summary table update doesn't need to run in trino, switching it to postgres
* Take ocpgcp sql changes from COST-444 OCPGCP unittest PR and break them out

Testing Instructions:
1. Spin koku up with trino, example: `make docker-up-min-trino`
2. Add an OCP on GCP source, for example with the test customer data: `make create-test-customer` followed by `make load-test-customer-data`
3. Hit a few of the OCP on GCP endpoints and verify the data exists, example: `http://localhost:8000/api/cost-management/v1/reports/openshift/infrastructures/gcp/costs/`
4. Connect to the DB and check the tables in the DB to be thorough and make sure the entries are in the different tables, example:

```
SELECT * FROM reporting_ocpgcp_compute_summary_p;
SELECT * FROM reporting_ocpgcp_cost_summary_p;
SELECT * FROM reporting_ocpgcp_cost_summary_by_account_p;
SELECT * FROM reporting_ocpgcp_cost_summary_by_gcp_project_p;
SELECT * FROM reporting_ocpgcp_cost_summary_by_region_p;
SELECT * FROM reporting_ocpgcp_cost_summary_by_service_p;
SELECT * FROM reporting_ocpgcp_database_summary_p;
SELECT * FROM reporting_ocpgcp_network_summary_p;
SELECT * FROM reporting_ocpgcp_storage_summary_p;
```


Optional way to check all the tables in one select statement to make sure they have data as an alternative to above, example output from below query:
```
 compute_count | cost_account_count | cost_gcp_project_count | cost_region_count | cost_service_count | cost_count | database_count | network_count | storage_count 
---------------+--------------------+------------------------+-------------------+--------------------+------------+----------------+---------------+---------------
            68 |                 34 |                     34 |                68 |                136 |         34 |             34 |            34 |            34
(1 row)
```
Example query that generates that output:
```
SELECT (
    SELECT COUNT(*)
    FROM reporting_ocpgcp_compute_summary_p
    ) as compute_count,
    (
    SELECT COUNT(*)
    FROM reporting_ocpgcp_cost_summary_by_account_p
    ) as cost_account_count,
    (
    SELECT COUNT(*)
    FROM reporting_ocpgcp_cost_summary_by_gcp_project_p
    ) as cost_gcp_project_count,
    (
    SELECT COUNT(*)
    FROM reporting_ocpgcp_cost_summary_by_region_p
    ) as cost_region_count,
    (
    SELECT COUNT(*)
    FROM reporting_ocpgcp_cost_summary_by_service_p
    ) as cost_service_count,
    (
    SELECT COUNT(*)
    FROM reporting_ocpgcp_cost_summary_p
    ) as cost_count,
    (
    SELECT COUNT(*)
    FROM reporting_ocpgcp_database_summary_p
    ) as database_count,
    (
    SELECT COUNT(*)
    FROM reporting_ocpgcp_network_summary_p
    ) as network_count,
    (
    SELECT COUNT(*)
    FROM reporting_ocpgcp_storage_summary_p
    ) as storage_count
;
```
---
Additional Context:
These don't need to run on trino and moving them to postgres will match the flow we have for all of our other providers
